### ISSUE TITLE
Weird patterns in simulated exposures

### ISSUE THREAD

**Issue by pierluigi-rinaldi** [spacetelescope/mirage#770]
I noticed weird patterns in simulated images (MIRAGE 2.1.0 + jwst 1.3.3).

In the first place, I thought it was a problem introduced by the JWST pipeline. However, digging a little deeper, I noticed that those strange patterns are introduced by MIRAGE (maybe caused by a mismatch between the reference files adopted by MIRAGE and those adopted by the pipeline).

You can find the most evident example of those weird patterns if you look at the A5 detector, where you can see a sort of 'hole' in the image (at the bottom of the simulated exposure after you run the first two stages of the pipeline).

Apart from that, I noticed vertical and horizontal stripes too.

Is there a chance to fix those problems? I am working on solutions to fix them, but I am afraid of introducing spurious effects on the final image I want to analyse.

Thanks in advance!

**Comment by bhilbert4** [comment]
Could you send a screen shot of these patterns? 

Mirage uses a dark current exposure from ground testing as the base of the simulation. This is done so that real detector effects like the bias structure, 1/f noise (horizontal lines), etc, are present in the simulated data, in order to make them as realistic as possible. So it may be that you are seeing real effects. But it would help to see an image of what you are referring to in order to be sure.

**Comment by x12red** [comment]
Here is an example. In this case, the image refers to the A5 detector.
It seems that the pipeline cannot get rid of those artefacts...

<img width="425" alt="A5_detector_screenshot" src="https://user-images.githubusercontent.com/74990883/156022018-75eee410-a2e5-4403-a20c-1af1cfc06e98.png">

**Comment by x12red** [comment]
Moreover, I wonder if I need to download again mirage reference files once I updated mirage from the previous version...

**Comment by bhilbert4** [comment]
No, you don't need to re-download the reference files. 

The bright spot in the bottom center of the image is a known artifact. Did you run the flat fielding step on these data? I would have expected that to remove this object.

The horizontal stripes are from residual 1/f noise. These will also appear in real data. Part of the 1/f noise is removed by the reference pixel subtraction step of the pipeline, but there is some left over afterwards. 

The big vertical stripe appears to be a bias offset. This is a rate image that you are showing right? Given the extra space around the edges, I'm assuming this is an i2d file? In that case a bias offset doesn't make a lot of sense as the cause. I would expect this affect to go away after the reference pixel subtraction step.

**Comment by x12red** [comment]
Thanks for your reply, Bryan!

This image is a screenshot from the final stacked image (i.e., after stage 3 of the pipeline), so the white space around the edge is simply due to the screenshot I took from DS9.

Yesterday I was able to remove both vertical and horizontal stripes. To do that, I applied an extra step between stage 1 and stage 2 of the pipeline. However, I do not know how to remove that artefact in the bottom centre of the image. I tried to produce some blank images (i.e., no sources, no CRs, bkg = 0) in order to create a sort of master flat field to apply after stage 2, but it did not work. Do you have any suggestions?

**Comment by bhilbert4** [comment]
Maybe try scaling and applying the flat field reference file to see if that helps? I'm still confused why it wasn't removed in the flat field step.

**Comment by x12red** [comment]
Yeah, also applying by hand that flat field reference file, I do not obtain any improvement...

**Comment by bhilbert4** [comment]
After speaking with some others, the darker vertical band makes more sense to me now. The 1/f noise is present in the reference pixels as well as the science pixels, and if it happens that the signal in the reference pixels is low due to the 1/f noise, then the bias level may be undersubtracted. The residual flat field feature is still a mystery to me though.
